azurerm/data_source_app_service_plan.go

```diff
@@ -86,21 +86,24 @@ func dataSourceAppServicePlanRead(d *schema.ResourceData, meta interface{}) erro
 	ctx := meta.(*ArmClient).StopContext
 	resp, err := client.Get(ctx, resourceGroup, name)
 	if err != nil {
-		if utils.ResponseWasNotFound(resp.Response) {
-			d.SetId("")
-			return nil
-		}
-
 		return fmt.Errorf("Error making Read request on App Service Plan %q (Resource Group %q): %+v", name, resourceGroup, err)
 	}
 
+	if utils.ResponseWasNotFound(resp.Response) {
```

Comment:
Does this change need to be rolled out to all of the resources? Or is this just app service plan specific?

Reply:
it appears to be resource dependent, unfortunately - so we'll need to go through the lot (although, that's outside the scope of this PR)